File: Sorting Algo/Basics-Bubble,Insertion,Selection/insertionSort.js
/**
 * Builds up the sort by gradually creating a larger left half which is always sorted
 */

function insertionSort(arr) {
    for (let i = 1; i < arr.length; i++) {
        let currentVal = arr[i];
        let j = i - 1
        for (; j >= 0 && arr[j] > currentVal; j--) {
            arr[j + 1] = arr[j];
        }
        arr[j + 1] = currentVal
    }

    return arr;
}

console.log(insertionSort([3, 2, 12, 4, 23, 543, 234, 1]))

File: Sorting_1/src/InsertionSort.java

public class InsertionSort {
	public static void printArray(int [] arr) {
		for(int i:arr) {
			System.out.print(i+" ");
		}
	}
	public static void insertionSort(int []arr) {
		int n=arr.length;
		int i=1;
		while(i<n) {
			int j=i-1;
			int curr=arr[i];
			while(j>=0 && arr[j]>curr ) {
				arr[j+1]=arr[j];
				j--;
			}
			arr[j+1]=curr;
			printArray(arr);
			System.out.println();
			i++;
		}


	}


	public static void main(String[] args) {

		int arr[]= {1,22,3,5,6,7,-1};
		insertionSort(arr);
		printArray(arr);



	}}


